org.metaborg.spoofax.shell.eclipse/src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseEditor.java

```diff
@@ -76,9 +89,14 @@ public void setFocus() {
      *
      * @return A new {@link Observable} from this editor.
      */
-    public Observable<String> asObservable() {
-        return Observable
-            .create((Observable.OnSubscribe<String>) EclipseEditor.this.observers::add);
+    public Observable<String> asObservable(boolean live) {
```

Comment:
I don't think there should be, but we could consider a framework for the client. Currently the only framework is that the client **must** implement an `IRepl` for the endpoint and an `IResultVisitor` to display results.

If the back end requires knowledge of an input component, or we want to provide a framework for that component to live in (like `IDisplay` for `IResultVisitor` is a framework for the output window), we could consider specifying an interface for "`IReplEditor`" as well. See also the createRepl factory method (`IWidgetFactory`).